docs/madr/decisions/022-retry-host-selection.md

```diff
@@ -0,0 +1,75 @@
+# Retry Host Selection
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/5897
+
+## Context and Problem Statement
+
+We want to add some additional retry fields plumbed through from Envoy to our Retry policy based on user demand. This is an HTTP only policy (no GRPC).
+
+```
+retry_priority
+retry_host_predicate
+host_selection_retry_max_attempts
+```
+
+### Design 
+
+Envoy has "host predicates" and "priority predicates" to enforce an additional logic 
+when selecting a host for the attempt. With `retry_host_predicate` we get:
+
+* don't retry on previously retried hosts
+* don't retry on canary hosts (the host should be preliminarily marked with `canary: true`)
+* don't retry on hosts with specified metadata key/value
+
+And with `retry_priority` we get:
+
+* don't retry on previous priorities
+
+These predicates can be combined into a single `hostSelection` section:
+
+```yaml
+hostSelection:
+  - predicate: OMIT_PREVIOUS_HOSTS
```

Comment:
Should these be CamelCase? We decided to use CamelCase for new enum values but I suppose this particular resource should remain consistent.

Reply:
Can you link an example? I just copied this from the new MeshRetry MADR.

Reply:
Ha the MADR wasn't updated but yes the policy in CamelCase (if you check the implem you'll see it's coherent)